app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -31,12 +33,45 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             return value;
         }
 
-        function handleFiltersUpdate(filters) {
+        function handleFiltersUpdate(filters, oldVal) {
             var activeFilters = angular.copy(PostFilters.getCleanActiveFilters(filters));
             FilterTransformers.rawFilters = angular.copy(filters);
             // Remove set filter as it is only relevant to collections and should be immutable in that view
             delete activeFilters.set;
-            $scope.activeFilters = _.mapObject(activeFilters, makeArray);
+            var savedSearchEntity = PostFilters.getModeEntity();
+            if ($scope.savedSearch && savedSearchEntity.id !== $scope.savedSearch.id) {
+                $scope.savedSearch = savedSearchEntity;
+            } else if (!$scope.savedSearch && PostFilters.getMode() === 'savedsearch') {
+                $scope.savedSearch = savedSearchEntity;
+            }
+
+            /**
+             * This handles the requirement to have saved search filters displayed in a different way
+             * from the rest of the filters.
+             *  - If there is no saved_search in the filters, continue with the normal filters (else)
+             *  - If there is a saved search:
+             *  - - set it in the scope : this is what we will use to display the saved search and its filters in the UI
+             *  - - get clean version (no defaults) of the saved search filters
+             *  - - get a clean activeFilters array that does not include the saved search filters. $scope.activeFilters: this
+             *  is the array we will be using to show the "extra" filters the user can set AFTER they selected  a saved search
+             *  (keep in mind that when a saved search is selected all filters are erased in favor of the saved search)
+             *  - - - Diffing rules: value from activeFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: if the value is a number or string, return Value because it is a priority (the one from filters not from saved search)
```

Comment:
everything is an array and can be compared with $difference in a sane way, I think??